onnx/defs/parser.cc

```diff
@@ -53,7 +54,36 @@ Status ParserBase::Parse(Literal& result) {
       }
     } else
       result.value = std::string(from + 1, next_ - from - 2); // skip enclosing quotes
-  } else if ((isdigit(nextch) || (nextch == '-'))) {
+  }
+
+  // Simplify the next if by consuming a possible negative sign.
+  if (nextch == '-') {
+    ++next_;
+  }
+
+  // Check for float literals that start with alphabet characters.
+  if (isalpha(*next_)) {
+    // Has to be a special float literal now: (-)*(nan|inf|infinity).
+    while (next_ < end_ && isalpha(*next_)) {
+      ++next_;
+    }
+    ONNX_TRY {
+      static_cast<void>(std::stof(std::string(from, next_ - from)));
+      result.type = LiteralType::FLOAT_LITERAL;
+      result.value = std::string(from, next_ - from);
+    }
+    ONNX_CATCH(...) {
+      ONNX_HANDLE_EXCEPTION([&]() {
+        // Rewind the parser if this is not a valid float literal.
+        next_ = from;
```

Comment:
Seems to me `return ParseError("Invalid literal")` would be better than returning `Status::OK()` ?

Reply:
Of course, a more descriptive error message would be better, I just used "Invalid literal" as a placeholder.